nts-pool-ke/src/lib.rs

```diff
@@ -572,14 +568,14 @@ async fn supported_algorithms_request(
 
 fn private_key_from_bufread(
     mut reader: impl BufRead,
-) -> std::io::Result<Option<rustls::PrivateKey>> {
+) -> std::io::Result<Option<rustls::pki_types::PrivateKeyDer<'static>>> {
     use rustls_pemfile::Item;
 
     loop {
         match rustls_pemfile::read_one(&mut reader)? {
```

Comment:
It should be possible to replace this manual `match` implementation with [`rustls_pemfile::private_key`](https://docs.rs/rustls-pemfile/2.0.0/rustls_pemfile/fn.private_key.html)